security/https-2way-authz/src/main/java/io/quarkus/ts/openshift/security/https/twoway/authz/CertificateRolesAugmentor.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.ts.openshift.security.https.twoway.authz;
+
+import io.quarkus.security.credential.CertificateCredential;
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.SecurityIdentityAugmentor;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.smallrye.mutiny.Uni;
+
+import javax.inject.Singleton;
+
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Quarkus doesn't have a declarative/configuration way to assign roles when using certificate authentication.
```

Comment:
Should be there an feature request for this?

Reply:
Perhaps. Not sure.

Reply:
I'm not sure either :) Code in this file looks quite specific to the structure of the keystore, especially `extractRoles` method.

Reply:
For this case I can imagine declarative mapping via application.properties like this
```
quarkus.http.ssl.certificate.role-mapping.cn.client=user
quarkus.http.ssl.certificate.role-mapping.cn.guest-client=guest
```
This would cover just some (simple) use-case for role mapping.
wdyt?

Reply:
WildFly supports this, both with legacy security and Elytron. In both cases, the  configuration is quite crazy. See:

- legacy security:
    - https://github.com/rhoar-qe/thorntail-test-suite/blob/master/protocols/https-2way-authz/src/main/resources/project-defaults.yml
    - https://github.com/rhoar-qe/thorntail-test-suite/blob/master/protocols/https-2way-authz/src/main/resources/roles.properties
- Elytron:
    - https://github.com/rhoar-qe/thorntail-test-suite/blob/master/protocols/https-2way-authz-elytron/src/main/resources/project-defaults.yml
    - https://github.com/rhoar-qe/thorntail-test-suite/blob/master/protocols/https-2way-authz-elytron/src/main/resources/roles.properties

Hence, I created this class in a generic way. The only thing you need to change is the `extractRoles` method.

Reply:
I filed https://github.com/quarkusio/quarkus/issues/11434

Reply:
Thanks!